### PR TITLE
[rewrite] More noise stats

### PR DESCRIPTION
**Pull request by han0110**

### CODE REVIEW

File: crypto/src/core/rlwe/test.rs [hunk 1]
```diff
@@ -466,14 +498,16 @@ fn noise_stats() {
...
+        let var_sk = param.sk_distribution.variance();
+        let var_noise_pk = var_noise;
+        let var_noise_ct_sk = var_noise;
+        let var_noise_ct_pk = ring_size * (var_u * var_noise_pk + var_sk * var_noise) + var_noise;
         assert_eq!(noise_ct_sk.mean().round(), 0.0);
-        assert!((noise_ct_sk.log2_std_dev() - var_noise.sqrt().log2()).abs() < 0.01);
+        assert!((noise_ct_sk.log2_std_dev() - var_noise_ct_sk.sqrt().log2()).abs() < 0.01);
         assert_eq!(noise_ct_pk.mean().round(), 0.0);
-        assert!(noise_ct_pk.log2_std_dev() < var_noise_ct_pk.sqrt().log2());
+        assert!((noise_ct_pk.log2_std_dev() - var_noise_ct_pk.sqrt().log2()).abs() < 0.2);
```
Comment (Janmajayamall) [inline crypto/src/core/rlwe/test.rs:512]: why is this 0.2 ?

File: crypto/src/core/rlwe/test.rs [hunk 2]
```diff
@@ -466,14 +498,16 @@ fn noise_stats() {
...
         let ring_size = param.ring_size as f64;
-        let var_sk = param.sk_distribution.variance();
         let var_noise = param.noise_distribution.variance();
-        let var_noise_ct_pk =
-            ring_size * var_sk * var_noise + (ring_size * var_sk + 1.0) * var_noise;
+        let var_u = param.u_distribution.variance();
+        let var_sk = param.sk_distribution.variance();
+        let var_noise_pk = var_noise;
+        let var_noise_ct_sk = var_noise;
+        let var_noise_ct_pk = ring_size * (var_u * var_noise_pk + var_sk * var_noise) + var_noise;
```
Comment (Janmajayamall) [inline crypto/src/core/rlwe/test.rs:508]: I'm not sure whether float64 is high enough precision for this calculation. 

A better idea will be to generate these tests using a sage script.

File: crypto/src/core/rlwe/test.rs (hunk 1 above)
Comment (han0110) [reply crypto/src/core/rlwe/test.rs:512]: I'm not sure tbh, the variance stats collected is not really stable, and it's roughly +-0.2

File: math/src/distribution.rs [hunk 3]
```diff
@@ -122,6 +122,17 @@ pub trait DistributionVariance {
     }
 }
 
+impl DistributionVariance for Range<u64> {
+    fn variance(self) -> f64 {
```
Comment (Janmajayamall) [inline math/src/distribution.rs]: Variance of distribution is B^2 / 12 only when values are uniformly distributed in range [-B/2, B/2). Are you sure that is the case here ?

Comment (han0110) [reply math/src/distribution.rs]: Removed in https://github.com/gausslabs/phantom-zone/pull/37/commits/4a4850e4bac1d64a41b5635b0cf2d39c01236e69

**Pull request merged**